bed/info.go

```diff
@@ -2,11 +2,53 @@ package bed
 
 import (
 	"fmt"
-	"github.com/vertgenlab/gonomics/dna"
+  "github.com/vertgenlab/gonomics/dna"
 	"github.com/vertgenlab/gonomics/fasta"
+	"log"
 )
 
-// UngappedRegionsFromFa: finds all regions outside gaps in a given fasta record.
+func MultiFaUngappedRegions(records []fasta.Fasta, chromName string, seqName string) []Bed {
+	var answer = make([]Bed, 0)
+	var seqNameIndex int = -1
+	var inRegion bool = false
+	var startRefPos, endRefPos int = 0, 0
+	var lastRefPos, lastAlnPos = 0, 0
+	var currAlnPos int = 0
+
+	//first we find the index of seqName
+	for i := 0; i < len(records); i++ {
+		if records[i].Name == seqName {
+			seqNameIndex = i
+			break
+		}
+	}
+
+	if seqNameIndex == -1 {
+		log.Fatalf("Error: seqName: %s, not found in records.\n", seqName)
+	}
+
+	for currAlnPos = 0; currAlnPos < len(records[0].Seq); currAlnPos++ {
+		if dna.DefineBase(records[seqNameIndex].Seq[currAlnPos]) && !inRegion {
+			inRegion = true
+			startRefPos = fasta.AlnPosToRefPosCounter(records[0], currAlnPos, lastRefPos, lastAlnPos)
+			lastRefPos, lastAlnPos = startRefPos, currAlnPos
+		} else if !(dna.DefineBase(records[seqNameIndex].Seq[currAlnPos])) && inRegion {
+			endRefPos = fasta.AlnPosToRefPosCounter(records[0], currAlnPos, lastRefPos, lastAlnPos)
```

Comment:
I think it shouldn't because in the multifa each sequence's index is by AlnPos not RefPos. When it's a gap in the reference sequence, sequence[alnPos] stores the value "dna.Gap"

Reply:
I think when alnPos is a gap in reference sequence, the ref position to the right (the first position to the first that is not a gap) is reported
```
func AlnPosToRefPosCounterSeq(record []dna.Base, AlnPos int, refStart int, alnStart int) int {
	if alnStart > AlnPos {
		refStart, alnStart = 0, 0 //in case the alnStart was improperly set (greater than the desired position, we reset the counters to 0.
	}
	for t := alnStart; t < AlnPos; t++ {
		if t == len(record) {
			log.Fatalf("Ran out of chromosome.")
		} else if record[t] != dna.Gap {
			refStart++
		}
	}
	return refStart
}
```